File: next/security/backend-ms1/src/main/java/ma/zs/stocky/service/impl/admin/security/RoleUserAdminServiceImpl.java
package ma.zs.stocky.service.impl.admin.security;


import ma.zs.stocky.bean.core.security.RoleUser;
import ma.zs.stocky.dao.criteria.core.security.RoleUserCriteria;
import ma.zs.stocky.dao.facade.core.security.RoleUserDao;
import ma.zs.stocky.dao.specification.core.security.RoleUserSpecification;
import ma.zs.stocky.service.facade.admin.security.RoleUserAdminService;
import ma.zs.stocky.zynerator.service.AbstractServiceImpl;
import ma.zs.stocky.zynerator.util.ListUtil;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.ArrayList;





import org.springframework.beans.factory.annotation.Autowired;

import ma.zs.stocky.service.facade.admin.security.UserAdminService ;
import ma.zs.stocky.bean.core.security.User ;
import ma.zs.stocky.service.facade.admin.security.RoleAdminService ;
import ma.zs.stocky.bean.core.security.Role ;

import java.util.List;
@Service
public class RoleUserAdminServiceImpl extends AbstractServiceImpl<RoleUser, RoleUserCriteria, RoleUserDao> implements RoleUserAdminService {







    public List<RoleUser> findByUserId(Long id){
        return dao.findByUserId(id);
    }
    public int deleteByUserId(Long id){
        return dao.deleteByUserId(id);
    }
    public long countByUserId(Long id){
        return dao.countByUserId(id);
    }
    public List<RoleUser> findByRoleId(Long id){
        return dao.findByRoleId(id);
    }
    public int deleteByRoleId(Long id){
        return dao.deleteByRoleId(id);
    }
    public long countByRoleAuthority(String authority){
        return dao.countByRoleAuthority(authority);
    }






    public void configure() {
        super.configure(RoleUser.class, RoleUserSpecification.class);
    }


    @Autowired
    private UserAdminService userService ;
    @Autowired
    private RoleAdminService roleService ;

    public RoleUserAdminServiceImpl(RoleUserDao dao) {
        super(dao);
    }

}

File: nest/security/backend-ms1/src/app/shared/security/service/RoleUserAdminServiceImpl.ts
import {Injectable, NotFoundException} from "@nestjs/common";

import {AbstractServiceImpl} from "src/app/zynerator/service/AbstractServiceImpl";

import {RoleUserDao} from "src/app/shared/dao/facade/core/security/RoleUserDao";
import {RoleUserCriteria} from "src/app/shared/dao/criteria/core/security/RoleUserCriteria";
import {RoleUser} from "src/app/shared/bean/core/security/RoleUser";
import {RoleUserDto} from "src/app/shared/dto/security/RoleUserDto";


@Injectable()
export class RoleUserAdminServiceImpl extends AbstractServiceImpl<RoleUser, RoleUserCriteria, RoleUserDao>{

    constructor(private readonly dao: RoleUserDao ,
    ) {
        super(dao);
    }

    async save(item: RoleUser): Promise<RoleUser> {
        const saved = await this.dao.save(item);
        return saved;
    }


    async update(item: RoleUser): Promise<RoleUser> {
        const saved = await this.dao.saveOrUpdate(item);
        return saved;
    }

    async updateMultiple(items: RoleUser[]): Promise<void> {
        if (items) {
            items.forEach(e => this.update(e))
        }
    }

    async  findAllOptimized(): Promise<RoleUserDto[]> {
        return this.dao.findAllOptimized();
    }

    async findAll(): Promise<RoleUser[]> {
        return this.dao.findAll();
    }

    async findById(id: number): Promise<RoleUser> {
        return this.dao.findById(id);
    }


    async deleteById(id: number): Promise<number> {
        const existing = await this.findById(id);
        if (!existing) {
            throw new NotFoundException(`Role user with ID ${id} not found.`);
        }
        const result = await this.dao.deleteById(existing.id);
        return result;
    }

    async deleteMultiple(items: RoleUser[]): Promise<RoleUser[]> {
        const deletedItems: RoleUser[] = [];
        for (const item of items) {
            await this.deleteById(item.id);
            deletedItems.push(item);
        }
        return deletedItems;
    }

    async findByUserId(id: number): Promise<RoleUser[]> {
        return this.dao.findByUserId(id);
    }

    async deleteByUserId(id: number): Promise<number> {
        return this.dao.deleteByUserId(id);
    }
    async findByRoleId(id: number): Promise<RoleUser[]> {
        return this.dao.findByRoleId(id);
    }

    async deleteByRoleId(id: number): Promise<number> {
        return this.dao.deleteByRoleId(id);
    }

    async findWithAssociatedLists(id: number): Promise<RoleUser> {
        const result = await this.dao.findById(id);
    return result;
    }

    async updateWithAssociatedLists(item: RoleUser): Promise<RoleUser> {
         return await this.dao.saveOrUpdate(item);
    }
}

